vendor/github.com/jesseduffield/gocui/task.go

```diff
@@ -25,16 +43,34 @@ func (self *TaskImpl) Done() {
 
 func (self *TaskImpl) Pause() {
 	self.withMutex(func() {
-		self.busy = false
+		self.pauseNoMutex()
 	})
 }
 
+func (self *TaskImpl) pauseNoMutex() {
+	self.busy = false
+	for _, t := range self.subTasks {
+		t.pauseNoMutex()
+	}
+}
+
 func (self *TaskImpl) Continue() {
 	self.withMutex(func() {
-		self.busy = true
+		self.continueNoMutex()
 	})
 }
 
+func (self *TaskImpl) continueNoMutex() {
+	self.busy = true
+	for _, t := range self.subTasks {
+		t.continueNoMutex()
+	}
+}
+
+func (self *TaskImpl) AddSubTask(subTask Task) {
+	self.subTasks = append(self.subTasks, subTask)
```

Comment:
maybe this needs to use self.withMutex?

Reply:
That's probably a good idea anyway (added in ca49635f4d96), but doesn't solve the problem.